optuna/samplers/_nsga2/crossover.py

```diff
@@ -0,0 +1,457 @@
+        [
+            numerical_transform.transform(
+                {
+                    param_key: parent.params[param_key]
+                    for param_key in numerical_search_space.keys()
+                }
+            )
+            for parent in parents
+        ]
+    )  # Parent individual with NUMERICAL_DISTRIBUTIONS parameter.
+    if crossover_name == "uniform":
+        child_params_array = _uniform(
+            parents_numerical_params_array[0],
+            parents_numerical_params_array[1],
+            rng,
+            swapping_prob,
+        )
+    elif crossover_name == "blxalpha":
+        alpha = 0.5
+        child_params_array = _blxalpha(
+            parents_numerical_params_array[0], parents_numerical_params_array[1], rng, alpha
+        )
+    elif crossover_name == "sbx":
+        if len(study.directions) == 1:
+            eta = 2
+        else:
+            eta = 20
+        child_params_array = _sbx(
+            parents_numerical_params_array[0],
+            parents_numerical_params_array[1],
+            rng,
+            numerical_distributions,
+            eta,
+        )
+    elif crossover_name == "vsbx":
+        if len(study.directions) == 1:
+            eta = 2
+        else:
+            eta = 20
+        child_params_array = _vsbx(
+            parents_numerical_params_array[0], parents_numerical_params_array[1], rng, eta
+        )
+    elif crossover_name == "undx":
+        sigma_xi = 0.5
+        sigma_eta = 0.35 / np.sqrt(len(parents_numerical_params_array[0]))
+        child_params_array = _undx(
+            parents_numerical_params_array[0],
+            parents_numerical_params_array[1],
+            parents_numerical_params_array[2],
+            rng,
+            sigma_xi,
+            sigma_eta,
+        )
+    elif crossover_name == "spx":
+        epsilon = np.sqrt(len(parents_numerical_params_array[0]) + 2)
+        child_params_array = _spx(parents_numerical_params_array, rng, epsilon)
+    else:
+        assert False
+
+    child_numerical_params = numerical_transform.untransform(child_params_array)
+    child_params.update(child_numerical_params)
+    return child_params
+
+
+def crossover(
+    crossover_name: str,
+    study: Study,
+    parent_population: Sequence[FrozenTrial],
+    search_space: Dict[str, BaseDistribution],
+    rng: np.random.RandomState,
+    swapping_prob: float,
+    dominates: Callable[[FrozenTrial, FrozenTrial, Sequence[StudyDirection]], bool],
+) -> Dict[str, Any]:
+
+    numerical_search_space: Dict[str, BaseDistribution] = {}
+    numerical_distributions: List[BaseDistribution] = []
+
+    for key, value in search_space.items():
+        if isinstance(value, _NUMERICAL_DISTRIBUTIONS):
+            numerical_search_space[key] = value
+            numerical_distributions.append(value)
+
+    if len(numerical_distributions) != 0:
+        numerical_transform = _SearchSpaceTransform(numerical_search_space)
```

Comment:
`numerical_transform` is created conditionally but is always referenced further down. We could ensure that this variable is defined perhaps as an Optional (i.e. possibly hold `None`).

Reply:
If `len(numerical_distributions)==0` in L144, the `numerical_transform` passed to the `try_crossover` function will be `None`. 
However, if `len(numerical_distributions)==0`, the function will return at L57 in the `try_crossover` function before using `numerical_transform`.

Reply:
> If len(numerical_distributions)==0 in L144, the numerical_transform passed to the try_crossover function will be None.

Is this really true? My comment was that this indeed should be the logic, while it currently isn't.

Reply:
I'm sorry. I understand.
I need to add to assign `None` to `numerical_transform` if `len(numerical_distributions)==0`.

Reply:
Exactly, and no worries, sorry if I wasn't clear.